core/src/main/java/org/elasticsearch/common/transport/TransportAddress.java

```diff
@@ -46,5 +46,9 @@
 
     boolean sameHost(TransportAddress other);
 
+    default boolean isLocalAddress() {
```

Comment:
we should we have a default false here? feels strange. Also it feels that isLocalAddress might be confusing since we have the `LocalTransportAddress` class. Maybe flip it and call it isPublicAddress?